### ISSUE TITLE
foreign table name is not resolved correctly when adding a new column with a foreign key

### ISSUE THREAD

**Issue by mtuncer** [citusdata/citus#2286]
Repro Steps
```sql
create table rr(a int primary key);
select create_reference_table('rr');
create table tt(a int, b int);
select create_distributed_table('tt', 'a');
alter table tt add column c int references rr(a) on delete set null;
```
Result
```
ERROR:  relation "rr" does not exist
CONTEXT:  while executing command on localhost:9700
```
Worker log has
```
ERROR:  relation "rr" does not exist
SELECT worker_apply_shard_ddl_command (102300, 'public', 'alter table tt add column c int references rr(a) on delete set null;')
```

**Comment by mtuncer** [comment]
When you set a foreign key on existing columns, it works as expected
```sql
alter table tt add constraint ttrra foreign key (a) references rr(a);
```
Corresponding worker query is
```sql
SELECT worker_apply_inter_shard_ddl_command (102397, 'public', 102368, 'public', 'alter table tt add constraint ttrra foreign key (a) references rr(a);')
```

**Comment by marcocitus** [comment]
This seems like something we should fix in 7.5
